### PR TITLE
normalize Call.Name to be lowercase

### LINKED ISSUE #2471: should Call.Name be normalized to lower case?
works:
```
select percentile(value, 95) from foo
```
doesn't work (`"error": "expected one argument for PERCENTILE()")`:
```
select PERCENTILE(value, 95) from foo
```

there seems to be quite a few places in the code that compare to the lowercase version of `Call.Name`.  would it be better to normalize `Call.Name` to be lowercase when it's created or even to store the original as well as a normalized?

### PR DESCRIPTION
**Pull request by neonstalwart**
replaces #2472 and fixes #2471

### CODE REVIEW

**Comment by neonstalwart** [comment]
@otoolep what do you think of this approach?  https://github.com/influxdb/influxdb/pull/2473/files#diff-bf7dc665271ba12b7ae77761b1ca4668R1872 is the significant part of the change.

**Comment by otoolep** [comment]
Thanks @neonstalwart.

This has to be OK'd by @toddboom and @pauldix

**Comment by toddboom** [comment]
yeah, i think this is good. +1

File: cmd/influxd/server_integration_test.go [hunk 1]
```diff
@@ -1000,17 +1000,17 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
...
 		{
 			name:     "fill with previous",
-			query:    `select mean(val) from "%DB%"."%RP%".fills where time >= '2009-11-10T23:00:00Z' and time < '2009-11-10T23:00:20Z' group by time(5s) fill(previous)`,
+			query:    `select mean(val) from "%DB%"."%RP%".fills where time >= '2009-11-10T23:00:00Z' and time < '2009-11-10T23:00:20Z' group by time(5s) FILL(previous)`,
 			expected: `{"results":[{"series":[{"name":"fills","columns":["time","mean"],"values":[["2009-11-10T23:00:00Z",4],["2009-11-10T23:00:05Z",4],["2009-11-10T23:00:10Z",4],["2009-11-10T23:00:15Z",10]]}]}]}`,
 		},
 		{
 			name:     "fill with none, i.e. clear out nulls",
-			query:    `select mean(val) from "%DB%"."%RP%".fills where time >= '2009-11-10T23:00:00Z' and time < '2009-11-10T23:00:20Z' group by time(5s) fill(none)`,
+			query:    `select mean(val) from "%DB%"."%RP%".fills where time >= '2009-11-10T23:00:00Z' and time < '2009-11-10T23:00:20Z' group by time(5s) FILL(none)`,
```
Comment (easyrasta) [inline cmd/influxd/server_integration_test.go]: May be better to mix uppercase and lower in test

Comment (neonstalwart) [inline cmd/influxd/server_integration_test.go]: I'm not sure what you mean. I updated the tests to use a combination of all upper and all lower. are you suggesting I should also try MixEd casing?  I don't think it's necessary.

**Pull request merged**